packages/adblocker-benchmarks/blockers/cliqz-base.js

```diff
@@ -36,4 +36,12 @@ module.exports = class Cliqz {
       type,
     })).match;
   }
+
+  matchDebug({ url, frameUrl, type }) {
+    return this.engine.match(Request.fromRawDetails({
+      url,
+      sourceUrl: frameUrl,
+      type,
+    })).filter || null;
```

Comment:
By default the original filter might not be kept around (but the engine will try to recreate it from internal state. If we know we are in debug mode we can specify it to the `FiltersEngine.parse` method above:
```js

  static parse(rawLists, enableCompression, debug) {
    return new Cliqz(FiltersEngine.parse(rawLists, {
      enableCompression,
      integrityCheck: false,
      loadCosmeticFilters: false,
      debug: debug === true,
    }));
  }
```

Reply:
Done.